9_collaborator_data.yml

```diff
@@ -208,12 +208,18 @@ targets:
       labels = I('subseg_id'),
       title = I('Full Network - Upstream & Downstream'))
 
+  9_collaborator_data/psu/prms_azhr.csv:
+    command: download.file(
+      destfile = target_name,
+      url = I('https://raw.githubusercontent.com/nhm-usgs/nhmparamdb_CONUS/master/azrh.csv'))
+
```

Comment:
I assume that this is the same `azrh` params that are in the model given that the parameters for our DRB model cutout was based on the `nhmparamdb` and we haven't altered the `azrh` params for any segment. But there is potential that we may modified these parameters based on work to improve stream shading (and also could potentially calibrate these params). In this case, the output from the PRMS-SNTemp model wouldn't reflect predictions using the `nhmparamdb` azrh, so I  think it would be better if in the future this pulled directly from the PRMS-SNTemp parameters that produced the output. Will need to do #42 first though

Reply:
That sounds like a great approach. Farshid may be interested in additional model parameters as well, so having a pipeline to pull from local PRMS-SNTemp would be ideal.

Reply:
One note - it looks like these azimuth angles range from 0 to 2pi radians. SNTemp has a somewhat strange azimuth convention, where values are defined from the N-S line and must range from -pi/2 radians to pi/2 radians:

>The segment azimuth angle, Ar, is the angle in the horizontal plane that the stream segment makes from the north–south line (fig. 2). This angle is defined facing south along the north–south line for streams in the northern hemisphere. Values of the azimuth angle, in radians, range from –π/2 to π/2 (equivalent to –90 degrees to 90 degrees). 
_Source: [Sanders, 2017](https://doimspp-my.sharepoint.com/:b:/r/personal/hcorson-dosch_usgs_gov/Documents/Stream_T_Modeling/Model_manuals/Sanders_2017.pdf?csf=1&web=1&e=sOu2x5) pg. 6_

Here's a visual from [Bartholow, 2000](https://doimspp-my.sharepoint.com/:b:/r/personal/hcorson-dosch_usgs_gov/Documents/Stream_T_Modeling/Model_manuals/Bartholow_2000.pdf?csf=1&web=1&e=N6WgI3) pg. 80:
![image](https://user-images.githubusercontent.com/54007288/84668374-2f283a80-aee9-11ea-8423-3e88b76974b4.png)

So if that convention still holds in PRMS-SNTemp, we will need to adjust these azimuth parameters, which would mean digging into how they were computed (since the angle might be defined from a different point, e.g., due east). Alternatively, it could make sense for me to include generation of the azimuth angles in my script to compute topographic shade.

Reply:
Ohhhhhhhh, thanks for pointing that out! I'm still confused about what/whether this would means for current and future code.

It sounds like we'll need to do some digging if we want to make any intellectual sense of these values, if we want to pass them to SNTemp, and/or if we want to use them to model light, right?

But for the purpose of a machine learning algorithm (which doesn't care about the range of a parameter (though it might care about extremely negative vs extremely positive), do you think the current convention (whatever it is) will suffice? My current code takes the [circular] mean of the azimuth angles to come up with one number per subbasin: 
https://github.com/USGS-R/delaware-model-prep/blob/3635fc7d5cb4ac452a87e32c8112f258436556b8/9_collaborator_data/src/subset_sntemp.R#L68-L74
It would be even better if this code weighted the reaches by length, but do you think the current implementation should at least be good enough to provide a consistent distinction among general subbasin angles?

Reply:
Hi Alison - yes I think that approach is reasonable for the purposes of a machine learning algorithm, since the computed azimuth for each subbasin will account for the variation in azimuth among the stream reaches in the basin, if not for the length of each stream reach. It would be interesting to see how weighting the stream reaches by length would change the circular mean -- I'm not familiar enough with how the reaches are segmented to know if the smaller headwater reaches would be overrepresented in this calculation. As for these azimuth values, you're correct that we'll just need to dig into and adjust them if and when they are passed to SNTemp.